File: 2-golang-artifacts/primes.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// Function to check if a number is prime
func isPrime(n int) bool {
	if n <= 1 {
		return false
	}
	for i := 2; i*i <= n; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

// Function to generate the first N prime numbers
func generatePrimes(count int) []int {
	primes := []int{}
	num := 2
	for len(primes) < count {
		if isPrime(num) {
			primes = append(primes, num)
		}
		num++
	}
	return primes
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Enter the number of primes to generate: ")
	input, _ := reader.ReadString('\n')
	input = strings.TrimSpace(input)
	count, err := strconv.Atoi(input)
	if err != nil {
		fmt.Println("Invalid input. Please enter a valid number.")
		return
	}

	primes := generatePrimes(count)
	fmt.Println("First", count, "prime numbers:")
	for _, prime := range primes {
		fmt.Print(prime, " ")
	}
	fmt.Println()
}

File: src/primes.ts
export function generatePrimes(n: number): number[] {
  const primes: number[] = [];
  let num = 2; // Start checking for primes from 2

  while (primes.length < n) {
      if (isPrime(num)) {
          primes.push(num);
      }
      num++;
  }

  return primes;
}

export function isPrime(num: number): boolean {
  if (num <= 1) return false;
  if (num <= 3) return true;

  if (num % 2 === 0 || num % 3 === 0) return false;

  for (let i = 5; i * i <= num; i += 6) {
      if (num % i === 0 || num % (i + 2) === 0) return false;
  }

  return true;
}